src/Form/SponsorshipLevelType.php

```diff
@@ -16,9 +17,9 @@ final class SponsorshipLevelType extends AbstractType implements DataMapperInter
 {
     private $currency;
 
-    public function __construct(string $currency)
+    public function __construct(Currency $currency)
     {
-        $this->currency = $currency;
+        $this->currency = $currency->getValue();
```

Comment:
Leave it as a Currency object and call the getValue method inside of the buildForm method.

Reply:
Not resolved yet

Reply:
such already done